File: LeetCode/1_Two_Sum/1_Two_Sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        cache = {}
        for i, num in enumerate(nums):
            cache[num] = i
        
        # cache = {nums[i]: i for i in range(0, len(nums))}
            
        for i, one in enumerate(nums):
            two = target - one
            j = cache.get(two)
            if j != None and i != j:
                return [i ,j]


nums = [3,3]
target = 6

s = Solution()

print(s.twoSum(nums, target))

File: LeetCode/1_Two_Sum/1_Two_Sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	cache := make(map[int]int)
	for i, one := range nums {
		cache[one] = i
	}
	for i, one := range nums {
		two := target - one
		j, ok := cache[two]
		if ok == true && i != j {
			return []int{i, j}
		}
	}
	return []int{0, 1}
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9

	result := twoSum(nums, target)
	fmt.Print(result)
}
